unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1295,31 +1295,49 @@ bool shouldRenderImageSynthesis
                 Camera camera = thirdPartyCameras.ToArray()[i];
                 cMetadata.thirdPartyCameraId = i;
 
-                cMetadata.position = camera.gameObject.transform.position;
-                cMetadata.rotation = camera.gameObject.transform.eulerAngles;
-
-                //agent relative third party camera metadata here
-                //if the camera is a child of the base agent, then return local space values
-                if (camera.GetComponentInParent<BaseAgentComponent>()) {
-                    cMetadata.agentPositionRelativeThirdPartyCameraPosition = camera
-                        .gameObject
-                        .transform
-                        .localPosition;
-                    cMetadata.agentPositionRelativeThirdPartyCameraRotation = camera
-                        .gameObject
-                        .transform
-                        .localEulerAngles;
+                //to be deprecated at some point, will be replaced by more descriptive worldRelativeThirdPartyCamera....
+                cMetadata.position = camera.transform.position;
+                cMetadata.rotation = camera.transform.eulerAngles;
+                //currently redundant data as it is the same as metadata.position/rotation, but more descriptive naming convention
+                cMetadata.worldRelativeThirdPartyCameraPosition = camera.transform.position;
+                cMetadata.worldRelativeThirdPartyCameraRotation = camera.transform.eulerAngles;
+
+                //grab this for parent or agent relative stuff
+                var worldSpaceCameraRotationAsQuaternion = camera.transform.rotation;
+
+                //this may be the same info as the agentPositionRelative values since ThirdPartyCameras often are children of the agent
+                //but in cases where a third party camera is attached to an arm joint or the like, this can be useful
+                if (camera.transform.parent != null) {
+                    cMetadata.parentObjectName = camera.transform.parent.name;
+
+                    cMetadata.parentPositionRelativeThirdPartyCameraPosition =
+                        camera.transform.localPosition;
+
+                    //get third party camera rotation as quaternion in parent space
+                    cMetadata.parentPositionRelativeThirdPartyCameraRotation =
+                        camera.transform.localEulerAngles;
```

Comment:
Rather than `parentPositionRelative...` can we change this to just `parentRelative...`? I.e.

```c#
parentPositionRelativeThirdPartyCameraPosition -> parentRelativeThirdPartyCameraPosition
parentPositionRelativeThirdPartyCameraRotation -> parentRelativeThirdPartyCameraRotation
```

Reply:
Ok! Just talked with rose about this to confirm that its still clear that the third party camera position is relative to the parent's position, and that the third party camera rotation is relative to the parent's rotation.

Note that there was a change previously to make `parentPositionRelativeThirdPartyCameraRotation` to `parent RotationRelativeThirdPartyCameraRotation` to make it explicit that the rotation can only be relative to rotation, and position can only be relative to position, but I think we can just do this in the description of the variables in the documentation rather than bloating the names.

Reply:
I suppose it's not clear to me what it would mean for a rotation to be relative a position?

Reply:
updated with https://github.com/allenai/ai2thor/commit/12d8e0548244938bb119e08703dfb722101b6f64

Reply:
> I suppose it's not clear to me what it would mean for a rotation to be relative a position?

Right, I think most people can intuit that position only makes sense relative to a position, and rotation to a rotation, so my original intent of being extremely explicit was probably overkill